src/server/mcp.ts

```diff
@@ -226,11 +214,13 @@ export class McpServer {
       return EMPTY_COMPLETION_RESULT;
     }
 
+    // @ts-ignore
     const field = prompt.argsSchema.shape[request.params.argument.name];
```

Comment:
This seems like a problem. The below line assumes that `prompt.argsSchema` is a zod type. Shouldn't this rely on the json-schema somehow (though it's a bit more complicated, it looks like `Completable` is an extension of zod that's internal.

At least it should be 
```suggestion
    const field = (prompt.argsSchema as ZodObject<ZodRawShape>)?.shape?.[request.params.argument.name];
```

And I guess standardised completable support could come later. Similar for the one below.